bindings/wasm/src/account/chain_state.rs

```diff
@@ -0,0 +1,67 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use wasm_bindgen::prelude::*;
+
+use identity::account::ChainState as ChainState_;
+
+use crate::tangle::WasmMessageId;
+
+#[wasm_bindgen(js_name = ChainState, inspectable)]
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
+pub struct WasmChainState(pub(crate) ChainState_);
+
+#[wasm_bindgen(js_class = ChainState)]
+impl WasmChainState {
+  #[wasm_bindgen(constructor)]
+  pub fn new() -> Self {
+    WasmChainState(ChainState_::new())
+  }
+
+  /// Returns the integration message id of the last published update.
+  ///
+  /// Note: [`MessageId`] has a built-in `null` variant that needs to be checked for.
+  #[wasm_bindgen(js_name = lastIntegrationMessageId)]
+  pub fn last_integration_message_id(&self) -> WasmMessageId {
+    (*self.0.last_integration_message_id()).into()
+  }
+
+  /// Returns the diff message id of the last published update.
+  ///
+  /// Note: [`MessageId`] has a built-in `null` variant that needs to be checked for.
+  #[wasm_bindgen(js_name = lastDiffMessageId)]
+  pub fn last_diff_message_id(&self) -> WasmMessageId {
+    (*self.0.last_diff_message_id()).into()
+  }
+
+  /// Sets the last integration message id and resets the
+  /// last diff message id to [`MessageId::null()`].
+  #[wasm_bindgen(js_name = setLastIntegrationMessageId)]
+  pub fn set_last_integration_message_id(&mut self, message: WasmMessageId) {
+    self.0.set_last_integration_message_id(message.into())
+  }
+
+  /// Sets the last diff message id.
+  #[wasm_bindgen(js_name = setLastDiffMessageId)]
+  pub fn set_last_diff_message_id(&mut self, message: WasmMessageId) {
+    self.0.set_last_diff_message_id(message.into())
+  }
```

Comment:
Discussion: we have previously used `string` wherever `MessageId` is needed in the bindings. Should we embrace exposing 1-1 types everywhere and thus update existing instances with `WasmMessageId` or continue using `string`? I do think we need to be consistent and not have `string` in some places and `WasmMessageId` in others.

Reply:
I agree about being consistent. I know sometimes it can be annoying, but I would prefer if we could have 1-1 types. That way we can make it more type-safe I guess.